test/extended/images/mirror.go

```diff
@@ -0,0 +1,367 @@
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	kapiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	kclientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+const podStartupTimeout = 3 * time.Minute
+const testPod1 = `kind: Pod
+id: oc-image-mirror-test-1
+apiVersion: v1
+metadata:
+  name: oc-image-mirror-test-1
+  labels:
+    name: oc-image-mirror-test-1
+spec:
+  containers:
+  - name: registry-1
+    image: docker.io/library/registry
+    env:
+    - name: REGISTRY_HTTP_ADDR
+      value: :5001
+    - name: REGISTRY_HTTP_NET
+      value: tcp
+    - name: REGISTRY_LOGLEVEL
+      value: debug
+    - name: REGISTRY_STORAGE
+      value: inmemory
+  - name: shell
+    image: openshift/origin:latest
+    command:
+    - /bin/sleep
+    - infinity
+`
+
+const testPod2 = `kind: Pod
+id: oc-image-mirror-test-2
+apiVersion: v1
+metadata:
+  name: oc-image-mirror-test-2
+  labels:
+    name: oc-image-mirror-test-2
+spec:
+  containers:
+  - name: registry-1
+    image: docker.io/library/registry
+    env:
+    - name: REGISTRY_HTTP_ADDR
+      value: :5001
+    - name: REGISTRY_HTTP_NET
+      value: tcp
+    - name: REGISTRY_LOGLEVEL
+      value: debug
+    - name: REGISTRY_STORAGE
+      value: inmemory
+  - name: registry-2
+    image: docker.io/library/registry
+    env:
+    - name: REGISTRY_HTTP_ADDR
+      value: :5002
+    - name: REGISTRY_HTTP_NET
+      value: tcp
+    - name: REGISTRY_LOGLEVEL
+      value: debug
+    - name: REGISTRY_STORAGE
+      value: inmemory
+  - name: shell
+    image: openshift/origin:latest
+    command:
+    - /bin/sleep
+    - infinity
+`
+
+func getRandName() string {
+	c := 20
+	b := make([]byte, c)
+
+	_, err := rand.Read(b)
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	return fmt.Sprintf("%02x", b)
+}
+
+func getCreds(oc *exutil.CLI) (string, string) {
+	user, err := oc.Run("whoami").Args().Output()
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	token, err := oc.Run("whoami").Args("-t").Output()
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	return strings.TrimSpace(user), strings.TrimSpace(token)
+}
+
+type testPod struct {
+	oc   *exutil.CLI
+	name string
+	spec string
+}
+
+func NewTestPod(oc *exutil.CLI, name, spec string) *testPod {
+	return &testPod{
+		oc:   oc,
+		name: name,
+		spec: spec,
+	}
+}
+
+func (pod *testPod) syncState(c kclientset.Interface, ns string, timeout time.Duration, state kapiv1.PodPhase) (err error) {
+	label := labels.SelectorFromSet(map[string]string{
+		"name": pod.name,
+	})
+
+	err = wait.Poll(2*time.Second, timeout,
+		func() (bool, error) {
+			podList, err := framework.WaitForPodsWithLabel(c, ns, label)
+			if err != nil {
+				framework.Failf("Failed getting pods: %v", err)
+				return false, nil // Ignore this error (nil) and try again in "Poll" time
```

Comment:
do you really want to Failf above then?  shouldn't it just log the error?

Reply:
yes. this is cut-and-paste error. Fixed.

https://github.com/openshift/origin/blob/master/test/extended/cluster/utils.go#L53